v0.12/alpine/entrypoint.sh

```diff
@@ -20,8 +9,20 @@ if [ -r $DEFAULT ]; then
     set +o allexport
 fi
 
-# chown home and data folder
-chown -R fluent /home/fluent
-chown -R fluent /fluentd
+# If the user has supplied only arguments append them to `fluentd` command
+if [ "${1:0:1}" = '-' ]; then
+    set -- fluentd "$@"
+fi
+
+# If user does not supply config file or plugins, use the default
+if [ "$1" == "fluentd" ]; then
+    if ! echo $@ | grep '\-c' ; then
```

Comment:
This seems even hackier than my solution.  Don't you think this will confuse users?

Reply:
By "hacky", I mean "fragile".  With the current solution, if there's the string `-c` _anywhere_ in the supplied arguments, it will not apply the default configuration, which is more likely to break.

For the purposes of the official images, I'm _much_ more concerned with the user's usage patterns than what the final generated command ends up being (multiple `-c` parameters would likely be slightly more confusing for users, but that can be trivially mitigated here by a comment that explains that the behavior of `fluentd` is to prefer the last-specified-flag for each option given, which is why these get pre-pended to the front of the command string instead of appended to the end).  The end goal is to have a reliable means of supplying a default configuration file and a trivial way for users to override it, which the solution I've proposed will provide in all cases I've tested, which I can't say about `echo ... | grep '\-c'`.  The `echo` solution would be slightly better with something like `haveConfig=; for arg; do if [ "$arg" = '-c' ]; then haveConfig=1; break; fi; done; if [ -z "$haveConfig" ]; then set -- "$@" -c ...; fi`, but is significantly more verbose.

I think this is likely another place to solicit preference from the maintainer (@repeatedly). :+1: